typegen/evaluation/file_type_hints_collector.py

```diff
@@ -0,0 +1,159 @@
+import os.path
+import pathlib
+from typing import Dict
+import pandas as pd
+import libcst as cst
+from libcst.metadata import PositionProvider
+import constants
+from tracing import TraceDataCategory
+
+
+class FileTypeHintsCollector:
+    """Collects the type hints of multiple .py files."""
+    def __init__(self, project_dir: pathlib.Path):
+        self.project_dir = project_dir
+        self.typehint_data = pd.DataFrame()
+        self.file_path = ""
+
+    def collect_data(self, file_paths: list[pathlib.Path]) -> None:
+        """Collects the type hints of the provided file paths."""
+        for file_path in file_paths:
+            with file_path.open() as file:
+                file_content = file.read()
+
+            module = cst.parse_module(source=file_content)
+            module_and_meta = cst.MetadataWrapper(module)
+            relative_path = str(pathlib.Path(os.path.relpath(file_path.resolve(), self.project_dir.resolve())))
```

Comment:
Use the `relative_to` method on `pathlib.Path` instances

Reply:
The relative_to does not work for example on "a/b/c/d" and "a/b/e", while the os path relpath does.

Reply:
that's because those paths are not relative to each other.
But you are only loading file from within the project directory, as these are the files we want to type hint, so this case cannot occur. If it does, that means a bug is somewhere else.

Reply:
Done